samples/mpu_stack_guard_test/src/main.c

```diff
@@ -10,61 +10,80 @@
 #include <misc/printk.h>
 
 /* size of stack area used by each thread */
-#define STACKSIZE 1024
+#define STACKSIZE  512
 
 /* scheduling priority used by each thread */
 #define PRIORITY 7
 
 /* delay between greetings (in ms) */
 #define SLEEPTIME 500
 
+/* delay between greetings (in ms) */
+#define SCHEDTIME 30
+
 /* Focaccia tastes better than DEEDBEEF ;-) */
 #define STACK_GUARD_CANARY 0xF0CACC1A
 
 struct stack_guard_buffer {
 	/* Make sure canary is not optimized by the compiler */
 	volatile u32_t canary;
 	struct k_thread thread;
+
 	u8_t __aligned(STACK_ALIGN) stack[STACKSIZE];
+
 };
 
-struct stack_guard_buffer buf;
+#define LAST_THREAD_NUM 3
+struct stack_guard_buffer buf[LAST_THREAD_NUM+1];
 
-u32_t recursive_loop(u32_t counter)
+u32_t recursive_loop(u32_t counter, int num, void *dummy)
 {
-	if (buf.canary != STACK_GUARD_CANARY) {
-		printk("Canary = 0x%08x\tTest not passed.\n", buf.canary);
+
+	if (buf[num].canary != STACK_GUARD_CANARY) {
+		printk("Canary = 0x%08x\tTest not passed.\n", buf[num].canary);
```

Comment:
I tried running this on frdm_k64f without stack guarding enabled and I don't hit this condition anymore.